javaslang/src/test/java/javaslang/collection/TraverableTest.java

```diff
@@ -0,0 +1,45 @@
+package javaslang.collection;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+
+public class TraverableTest {
+
+    // -- max()
+
+    @Test
+    public void maxShouldBeEmpty() {
+         assertTrue(Iterator.empty().max().isEmpty());
+    }
+
+    @Test
+    public void maxShouldBeTheOnlyValue() {
+        assertEquals(Integer.valueOf(5), Iterator.of(5).max().get());
+    }
+
+    @Test
+    public void maxShouldBeTheMaxValue() {
+        assertEquals(Integer.valueOf(3), Iterator.of(2, 3).max().get());
+    }
```

Comment:
We already have that test, it can be deleted:

```java
@Test
public void shouldComputeMaxOfInt() {
    assertThat(of(1, 2, 3).max()).isEqualTo(Option.some(3));
}
```